crates/proto/src/rr/rdata/caa.rs

```diff
@@ -844,20 +844,18 @@ impl fmt::Display for Value {
 
         match self {
             Value::Issuer(name, values) => {
-                match name {
-                    Some(name) => write!(f, "{}", name)?,
-                    None => write!(f, ";")?,
+                if let Some(name) = name {
+                    write!(f, "{}", name)?;
                 }
-
-                if let Some(value) = values.first() {
-                    write!(f, " {}", value)?;
-                    for value in &values[1..] {
-                        write!(f, "; {}", value)?;
-                    }
+                for value in values.iter() {
+                    write!(f, "; {}", value)?;
                 }
             }
             Value::Url(url) => write!(f, "{}", url)?,
-            Value::Unknown(v) => write!(f, "{:?}", v)?,
+            Value::Unknown(v) => match str::from_utf8(v) {
+                Ok(text) => write!(f, "{}", text)?,
+                Err(_) => return Err(fmt::Error),
+            },
```

Comment:
I've changed here so non-utf8 bytes will raise an error.
Since the parser will take as input a `&str`, the error arm of this match should be unreachable.
Therefore, I'm making this fail loudly, instead of producing a wrong value.

Reply:
@bluejekyll sorry, I added this one extra change.